File: babylonjs/src/Show/ShowType/redShow.ts
import {Show} from "../show";
import {Color3, Vector3} from "@babylonjs/core";

export class RedShow extends Show {
    private static SPEED: number = 0.055;
    private static RADIUS: number = 2.0;

    protected ballCount(): number {
        return 1;
    }

    protected getColor(): Color3 {
        return new Color3(1, 0.1, 0.1);
    }

    protected anim(): void {
        const t = this._time * RedShow.SPEED;

        const x = Math.cos(t) * RedShow.RADIUS;
        const z = Math.sin(t) * RedShow.RADIUS;
        this._balls[0].position = new Vector3(x, 1.8, z);
    }
}

File: unity/Assets/Show/ShowType/RedShow.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace FiveButtons
{
    public class RedShow : Show
    {
        private const float SPEED = 0.01f;
        private const float RADIUS = 1.0f;
        
        protected override Color GetColor() => new Color(0.88f, 0.0f, 0.0f);

        protected override void Anim()
        {
            var t = _time * SPEED;
            
            var x = Mathf.Cos(t) * RADIUS;
            var z = Mathf.Sin(t) * RADIUS;
            _balls[0].transform.localPosition = new Vector3(x, 1.8f, z);
        }
    }
}